File: app/src/main/java/com/datarockets/mnchkn/activities/chooser/ChooseGamePresenterImpl.java
package com.datarockets.mnchkn.activities.chooser;

public class ChooseGamePresenterImpl implements ChooseGamePresenter {

    private ChooseGameView mChooseGameView;
    private ChooseGameInteractor mInteractor;

    public ChooseGamePresenterImpl(ChooseGameView chooseGameView) {
        this.mChooseGameView = chooseGameView;
        this.mInteractor = new ChooseGameInteractorImpl();
    }

    @Override
    public void onResume() {
    }

    @Override
    public void onDestroy() {
        if (mChooseGameView != null) {
            mChooseGameView = null;
        }
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/chooser/ChooseGamePresenterImpl.kt
package com.datarockets.mnchkn.activities.chooser

class ChooseGamePresenterImpl(private var mChooseGameView: ChooseGameView?) : ChooseGamePresenter {
    private val mInteractor: ChooseGameInteractorImpl

    init {
        this.mInteractor = ChooseGameInteractorImpl()
    }

    override fun onResume() {
    }

    override fun onDestroy() {
        if (mChooseGameView != null) {
            mChooseGameView = null
        }
    }

}
